codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Message.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http2;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+
+public abstract class Http2Message {
+    Http2Message() { }
```

Comment:
I've considered doing that, but I don't think I'm to the point I could guarantee a stable API for doing so. For example, I'm using stream id at the moment, but I think to support outgoing streams I'll need to change it to a "stream object" (but not Http2Stream). I'm trying to keep the structure very closed at the moment and open it up as able.

It seems like the "Netty way" allows people to create new messages, but those new messages can't behave any differently from existing messages. I'm strongly considering removing the `process()` call and just having a if-else/switch for each supported message type, as that seems to be the normal Netty pattern. That would allow making these messages into interfaces and then having default implementations.